### PR TITLE
Part c

### PR DESCRIPTION
**Pull request by sobel121**

### CODE REVIEW

File: domElementsHandler.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    task.appendChild(checkboxAndText);
+    task.appendChild(editAndDeleteButtons);
+
+    return task;
+};
+
+const createTaskObjectFromElement = (taskElement) => {
+    return taskElement === undefined ? taskElement : {id: taskElement.id, 
+        description: taskElement.children[0].children[1].value,
+        list: taskElement.children[0].children[0].checked ? "done" : "todo"};
```
Comment (odev954) [inline domElementsHandler.js]: utilsssssssssssssssssssssssssssssss

File: localStorageHandler.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    setTasksInLocalStorage();
+};
+
+const createTaskElementFromLocalStorage = (task) => {
+    const taskElement = createNewTaskElement(task.id, task.description);
+    setTaskList(task, taskElement);    
+};
+
+const initializeTasksFromLocalStorage = () => {
+
```
Comment (odev954) [inline localStorageHandler.js]: empty line

File: localStorageHandler.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    setTasksInLocalStorage();
+};
+
+const updateTaskDescriptionInLocalStorage = (event) => {
+    const taskId = getTaskElementFromHisChildren(event.target).id;
+    const taskIndex = tasks.findIndex(task => task.id === taskId);
+    
+    tasks[taskIndex].description = event.target.value;
+    event.target.dataset.recentTaskDescription = event.target.value;
```
Comment (odev954) [inline localStorageHandler.js]: put `event.target` in variable

File: localStorageHandler.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+const removeTaskFromLocalStorage = (event) => {
+    const taskId = getTaskElementFromHisChildren(event.target).id;
+
+    tasks = tasks.filter(task => task.id !== taskId);
+
+    setTasksInLocalStorage();
+};
+
+const updateTaskDescriptionInLocalStorage = (event) => {
+    const taskId = getTaskElementFromHisChildren(event.target).id;
```
Comment (odev954) [inline localStorageHandler.js]: >put `event.target` in variable

as  well

File: script.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    
+    todoList.appendChild(task);
+    addTaskToLocalStorage(task);
+    updateTaskNumberInLocalStorage();
+    clearTaskInput(event);
+};
+
+const getTaskDescriptionFromTask = (taskElement) => taskElement.children[0].children[1];
+
+const getTaskElementFromHisChildren = (element) => {
```
Comment (odev954) [inline script.js]: check if you can solve the hover problem without searching the task div when hovering children.

File: script.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            const afterTask = getDragAfterElement(list, dragged, event.clientY);
+            if (list === dragged.parentElement) {
+                if (afterTask === undefined) {
+                    list.appendChild(dragged);
+                } else {
+                    list.insertBefore(dragged, afterTask);
+                }
+
+                list.dragged = createTaskObjectFromElement(dragged);
+                list.afterTask = createTaskObjectFromElement(afterTask);
```
Comment (odev954) [inline script.js]: move this to separate function

File: script.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        list.addEventListener("dragend", () => {
+            updateLocalStorageAfterPositionSwap(list.dragged, list.afterTask)
+        });
+    });
+};
+
+setListsEventListeners();
+
+const getDragAfterElement = (list, dragged, elementHeight) => {
+    const draggableElements = Object.values(list.children).filter(task => task.id !== dragged.id);
```
Comment (odev954) [inline script.js]: move to utils

File: script.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+        list.addEventListener("dragend", () => {
+            updateLocalStorageAfterPositionSwap(list.dragged, list.afterTask)
+        });
+    });
+};
+
+setListsEventListeners();
+
+const getDragAfterElement = (list, dragged, elementHeight) => {
```
Comment (odev954) [inline script.js]: rename this function

File: script.js [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+    const draggableElements = Object.values(list.children).filter(task => task.id !== dragged.id);
+
+    return draggableElements.reduce((closest, currentTask) => {
+        const ElementBox = currentTask.getBoundingClientRect();
```
Comment (odev954) [inline script.js]:
```suggestion
        const elementBox = currentTask.getBoundingClientRect();

```

File: script.js [hunk 10]
```diff
@@ -0,0 +1,105 @@
+const todoList = document.getElementById("todo");
+const doneList = document.getElementById("done");
+const lists = [todoList, doneList];
+let taskNumber = getTaskNumberFromLocalStorage()
+let tasks = getTasksFromLocalStorage();
+
```
Comment (odev954) [inline script.js:6]: move all utillities to `utils.js`

**Review by odev954 — CHANGES_REQUESTED** [review]

File: localStorageHandler.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+    tasks[taskIndex].list = event.target.checked ? "done" : "todo";
+
+    placeMovedTaskLastInArray(taskIndex);
+
+    setTasksInLocalStorage();
+};
+
+const setTasksInLocalStorage = () => localStorage.setItem("tasks", JSON.stringify(tasks));
+
+const RemoveAllListTasksFromLocalStorage = (listName) => {
```
Comment (itayboop) [inline localStorageHandler.js]: why is it PascalCase?

**Review by itayboop — APPROVED** [review]

**Pull request merged**